crates/cli-support/src/js/mod.rs

```diff
@@ -318,12 +318,29 @@ impl<'a> Context<'a> {
         // Deno removed support for .wasm imports in https://github.com/denoland/deno/pull/5135
         // the issue for bringing it back is https://github.com/denoland/deno/issues/5609.
         format!(
-            "const file = new URL(import.meta.url).pathname;
-            const wasmFile = file.substring(0, file.lastIndexOf(Deno.build.os === 'windows' ? '\\\\' : '/') + 1) + '{}_bg.wasm';
-            const wasmModule = new WebAssembly.Module(Deno.readFileSync(wasmFile));
+            "const url = new URL(import.meta.url)
+
+            let wasmCode = ''
+            if (url.protocol.includes('file')) {{
+                let file = new URL(import.meta.url).pathname;
+                if (Deno.build.os === 'windows') {{
+                    if (file.startsWith('/')) file = file.substr(1)
+                    file = Deno.realPathSync(file + '/../{module_name}_bg.wasm')
+                }} else {{
+                    file = file.substring(0, file.lastIndexOf('/') + 1) + '{module_name}_bg.wasm'
+                }}
+                wasmCode = Deno.readFileSync(file)
+            }} else if (url.protocol.includes('http')) {{
+                const wasm_url = import.meta.url.substring(0, import.meta.url.lastIndexOf('/') + 1) + '{module_name}_bg.wasm'
+                wasmCode = await (await fetch(wasm_url)).arrayBuffer()
+            }} else {{
+                console.error(`Unsupported protocol: ${{url.protocol}}`)
+                Deno.exit(0)
+            }}
+            const wasmModule = new WebAssembly.Module(wasmCode);
```

Comment:
You should almost never use synchronous constructors, as they can be very slow for large Wasm. Instead, use `await WebAssembly.instantiate(wasmCode, imports)`.

Reply:
^ this issue still stands. (I'm not a maintainer, but I don't think we should ever block)

Reply:
ok! https://github.com/rustwasm/wasm-bindgen/pull/2297/commits/f71515d2c7bf601b80df832388fdabccf70b926f